app/process/handler/crypto-transfer/crypto-transfer.go

```diff
@@ -97,6 +97,12 @@ func (cth *CryptoTransferHandler) Handle(payload []byte) {
 			cth.logger.Errorf("Failed to create a transaction record for TransactionID [%s]. Error [%s].", ctm.TransactionId, err)
 			return
 		}
+
+		err = cth.transactionRepo.UpdateStatusInitial(ctm.TransactionId)
```

Comment:
Why is the `transaction` not directly created with `StatusInitial`?

Reply:
This question came through my mind as well. Initially I thought that the PENDING status is mandatory at this stage. I will change the flow now.